### PR TITLE
Remove the `inlineMathSQRT()` function on X86

### PR DESCRIPTION
**Pull request by simonameng**
Removing `inlineMathSQRT()` function on X86, because we are implementing `dsqrtEvaluator` on X86, which `StrictMath.sqrt()` and `Math.sqrt()` will be reduced to instead of calling `inlineMathSQRT()`.

The PR of `dsqrtEvaluator`'s implementation: https://github.com/eclipse/omr/pull/4514

### CODE REVIEW

**Comment by andrewcraik** [comment]
@simonameng I have marked this WIP until the OMR improvement is merged to prevent an accidental merge

File: test/functional/JIT_Test/playlist.xml [hunk 1]
```diff
@@ -393,4 +393,30 @@
 			<impl>ibm</impl>
 		</impls>
 	</test>
+
+<!-- jit.test.recognizedMethod tests start here -->
+	<test>
+		<testCaseName>jit_recognizedMethod</testCaseName>
+		<variations>
+			<variation>-Xjit:count=1,disableAsyncCompilation,dontInline={*TestJavaLang*}</variation>
```
Comment (fjeremic) [inline test/functional/JIT_Test/playlist.xml]: I think we need an `-Xint` variation here as well to validate the test itself is correct. Is there a reason we have `dontInline` in here? These sorts of things tend to break when refactoring happens as we omit to change the names here.

File: test/functional/JIT_Test/testng.xml [hunk 2]
```diff
@@ -548,5 +548,13 @@
       <class name="jit.test.vich.ReadTest" />
     </classes>
   </test>
+
+  <!-- jit.test.recognizedMethod tests start here -->
+  <test name="RecognizedSqrtTest">
```
Comment (fjeremic) [inline test/functional/JIT_Test/testng.xml]: We can generalize this name to something like "RecognizedMethodTests", since this group will have other recognized methods added to it moving forward. Do you agree that we should generalize the name?

File: test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangMath.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+public class TestJavaLangMath {
+
+    /*
+    * The following test is for testing the java_lang_Math_sqrt function. 
+    * In this test, we are expecting the calling to the function to be transformed into a dsqrt node in recognizedCallTransformer.
+    * Also, if the input of the function is a double type constant, we should see the dsqrt node being reduced to dconst in treeSimplification.
+    * The above transformation and reduction will be performed when the function get executed twice, therefore, we set the invocationCount to 2.
+    * The test include the corner cases that sqrt requires in document.
+    */
```
Comment (fjeremic) [inline test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangMath.java]: The documentation is a good start but it could use some improvements. We need to use Javadoc for Java documentation. We should also try to explain the _why_ and the _how_. For example the first sentence of the documentation is not very useful as this is quite obvious from the method name. The comment about the `iternationCount` is good though as it tells us some things that are definitely non-obvious.

Let's try to improve and format:

```
/**
 * Tests the constant corner cases defined by the {@link Math.sqrt} method.
 * <p>
 * The JIT compiler will transform calls to {@link Math.sqrt} within this test
 * into the following tree sequence:
 *
 * <code>
 * dsqrt
 *   dconst <x>
 * </code>
 *
 * Subsequent tree simplification passes will attempt to reduce this constant
 * operation to a <code>dsqrt<code> IL by performing the square root at compile
 * time. However we must ensure the result of the square root done by the
 * compiler at compile time will be exactly the same as the result had it been
 * done by the Java runtime at runtime. This test validates the results are the
 * same.
 */
```

Comment (fjeremic) [reply test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangMath.java]: Similar change needed for the other file.

File: test/functional/JIT_Test/playlist.xml (hunk 1 above)
Comment (simonameng) [reply test/functional/JIT_Test/playlist.xml]: > I think we need an -Xint variation here as well to validate the test itself is correct.
Do you mean adding another test with the variation '-Xint'?

> Is there a reason we have dontInline in here?
I add the option because the function would be inlined in my previous test and I couldn't generate the log of the functions, so I add it just in case. But I think we don't need the `dontInline` here. I tried to run the command without `dontInline` option and the functions don't get inlined either. I am going to remove it.

File: test/functional/JIT_Test/testng.xml (hunk 2 above)
Comment (simonameng) [reply test/functional/JIT_Test/testng.xml]: Agree. I am going to change it.

File: test/functional/JIT_Test/playlist.xml (hunk 1 above)
Comment (fjeremic) [reply test/functional/JIT_Test/playlist.xml]: > Do you mean adding another test with the variation '-Xint'?

That's right.

File: test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangStrictMath.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    * the same as the result had it been done by the Java runtime at runtime. This 
+    * test validates the results are the same.
+    */
+    @Test(groups = {"level.sanity"}, invocationCount=2)
+    public void test_java_lang_StrictMath_sqrt() {
+        AssertJUnit.assertTrue(Double.isNaN(StrictMath.sqrt(Double.NEGATIVE_INFINITY)));
+        AssertJUnit.assertTrue(Double.isNaN(StrictMath.sqrt((double)-42.25)));
+        AssertJUnit.assertEquals(-0.0, StrictMath.sqrt((double)-0.0));
+        AssertJUnit.assertEquals(+0.0, StrictMath.sqrt((double)+0.0));
+        AssertJUnit.assertEquals(7.5, StrictMath.sqrt((double)56.25));
```
Comment (fjeremic) [inline test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangStrictMath.java]: Can we change these to explicit double constants? i.e. remove `(double)` and `d` suffix to the floating point numbers:

```
        AssertJUnit.assertTrue(Double.isNaN(StrictMath.sqrt(-42.25d)));
        AssertJUnit.assertEquals(-0.0d, StrictMath.sqrt(-0.0d));
        AssertJUnit.assertEquals(+0.0d, StrictMath.sqrt(+0.0d));
        AssertJUnit.assertEquals(7.5d, StrictMath.sqrt(56.25d));
```

Comment (fjeremic) [reply test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangStrictMath.java]: Similarly below.

Comment (simonameng) [reply test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangStrictMath.java]: Changed

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8 depends eclipse/omr#4514

**Comment by fjeremic** [comment]
@andrewcraik this change is dependent on the OMR change which was merged recently. The OMR-Acceptance pipeline will block until this is merged. It may be a good idea to merge this sometime tonight so the short build break doesn't affect people.

**Comment by fjeremic** [comment]
@andrewcraik anything left for this change?
